src/App.jsx

```diff
@@ -1,28 +1,51 @@
 import './App.css';
-// import AddRestaurantModal from './components/AddRestaurantModal';
 import CategoryFilter from './components/CategoryFilter';
 import Header from './components/Header';
-import RestaurantDetailModal from './components/RestaurantDetailModal';
 import RestaurantList from './components/RestaurantList';
-import { useState } from 'react';
+import Modal from './components/Modal';
+import AddRestaurantModal from './components/AddRestaurantModal';
+import RestaurantDetailModal from './components/RestaurantDetailModal';
+import { useEffect, useState } from 'react';
 import { restaurantsData } from './data/restaurantsData';
 
 function App() {
+    const [restaurants, setRestaurants] = useState(restaurantsData);
     const [category, setCategory] = useState('전체');
-    const filteredRestaurants = category === '전체' ? restaurantsData : restaurantsData.filter((restaurant) => restaurant.category === category);
     const [isModalOpen, setIsModalOpen] = useState(false);
     const [selectedRestaurant, setSelectedRestaurant] = useState(null);
+    const [modalKind, setModalKind] = useState('');
+
+    useEffect(() => {
+        if (category === '전체') {
+            setRestaurants(restaurantsData);
+        } else {
+            const filteredRestaurants = restaurantsData.filter((restaurant) => restaurant.category === category);
+            setRestaurants(filteredRestaurants);
+        }
+    }, [category]);
```

Comment:
```suggestion
    useEffect(() => {
        if (category === '전체') {
            setRestaurants(restaurantsData);
            return;
        }
        
        const filteredRestaurants = restaurantsData.filter((restaurant) => restaurant.category === category);
        setRestaurants(filteredRestaurants);
    }, [category]);
```
저는 주로 `else`문을 사용하기 보다는 `early return` 문법을 사용하는 것 같은데 세오는 `else`문을 사용하시는 편이신가요??😮